### ISSUE TITLE
static build is broken

### ISSUE THREAD

**Issue by robx** [PostgREST/postgrest#2327]
It seems the static-haskell-nix hack from #2292 produces non-static binaries:

Unfortunately, I no longer seem to be able to produce a static binary:

```
nix-build --attr postgrestStatic

ldd result/bin/postgrest
        linux-vdso.so.1 (0x00007ffe5bb03000)
        libz.so.1 => /nix/store/srww288f7sb0f8hb5399431k8ldg2pss-zlib-1.2.11/lib/libz.so.1 (0x00007fda1a070000)
        libpq.so.5 => /nix/store/8y5nh8zx42qbzw0ki5rc9ywgi7z3axqb-postgresql-14.2-lib/lib/libpq.so.5 (0x00007fda19d2c000)
        libgmp.so.10 => /nix/store/wxyn3pv3kvqrl2lnfzpaa82vsd6fizka-gmp-6.2.1/lib/libgmp.so.10 (0x00007fda19c8c000)
        libc.so => /nix/store/88nqrkcal1mmiqbx10ax6wkyncf1h1gk-musl-1.2.2/lib/libc.so (0x00007fda19bd9000)
        libffi.so.8 => /nix/store/wb3ji9r5b2f747phif5lfx7np45fl9px-libffi-3.4.2/lib/libffi.so.8 (0x00007fda19bcc000)
```

[9.0.1](https://github.com/PostgREST/postgrest/releases/tag/v9.0.1) shows:

```
ldd postgrest
$       not a dynamic executable
```

We need a test on CI to ensure the above output.

---

On another NixOS server, trying to run produces an error(`No such file or directory`) and when running ldd on it:

```
ldd /nix/store/hpx1rapsd6ylshh8d115ab7ifpjd3bia-postgrest/bin/postgrest

/nix/store/hpx1rapsd6ylshh8d115ab7ifpjd3bia-postgrest/bin/postgrest: 
error while loading shared libraries: 
/nix/store/2pi6zgkwnr3zdslvlv16nixpzvbyjx1n-glibc-2.31/lib/libc.so: invalid ELF header
```

_Originally posted by @steve-chavez in https://github.com/PostgREST/postgrest/issues/2292#issuecomment-1155884004_

**Comment by robx** [comment]
Likeliest thing to look into is this patch: https://github.com/PostgREST/postgrest/blob/main/nix/patches/static-haskell-nix-isexecutable.patch. If we're lucky, that test is just wrong, and if we fix it we get a static build.

If not, it's a bit less clear how to proceed. https://github.com/nh2/static-haskell-nix is quite outdated and feels a bit abandoned, and it seems @monacoremo who introduced this here is not around anymore either?

**Comment by steve-chavez** [comment]
> If not, it's a bit less clear how to proceed. https://github.com/nh2/static-haskell-nix is quite outdated and feels a bit abandoned

Yeah, that is is surprising, no issues opened in 2022 too. Seeing https://github.com/NixOS/nixpkgs/issues/43795#issuecomment-1092029035, perhaps `static-haskell-nix` is not needed anymore for static binaries?

**Comment by steve-chavez** [comment]
From https://github.com/NixOS/nixpkgs/issues/61575#issuecomment-1114159373

>  don't know about other use cases, but for TemplateHaskell we seems to be doing fine with just -fexternal-dynamic-refs. On the unstable channel pkgsStatic.haskellPackages can build pretty complex applications fully statically, so this may not be necessary anymore.

**Comment by robx** [comment]
Ooh `pkgsStatic.haskellPackages` looks interesting! For now, I do think I managed to fix things in #2328.
